cert/tls.go

```diff
@@ -24,17 +26,38 @@ var (
 	}
 )
 
+type TlsReloader struct {
+	certMu sync.RWMutex
+	cert   *tls.Certificate
+}
+
+// GetCertBytesFromPath reads the TLS certificate and key files at the given
+// certPath and keyPath and returns the file bytes.
+func GetCertBytesFromPath(certPath, keyPath string) (certBytes,
+	keyBytes []byte, err error) {
+
+	certBytes, err = ioutil.ReadFile(certPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	keyBytes, err = ioutil.ReadFile(keyPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	return certBytes, keyBytes, nil
+}
+
 // LoadCert loads a certificate and its corresponding private key from the PEM
-// files indicated and returns the certificate in the two formats it is most
+// bytes indicated and returns the certificate in the two formats it is most
 // commonly used.
-func LoadCert(certPath, keyPath string) (tls.Certificate, *x509.Certificate,
+func LoadCert(certBytes, keyBytes []byte) (tls.Certificate, *x509.Certificate,
```

Comment:
Would it make sense to leave `LoadCert` with the old path signature, in order to keep it backwards compatible for projects that do not want to switch to the `CertReloader`?

Reply:
Ah makes sense. I think we need a function that takes in the bytes, though... Maybe it would make sense to leave LoadCert as is and create a second function LoadCertFromBytes with the new functionality?